src/sctools/fastq.py

```diff
@@ -0,0 +1,197 @@
+from collections import namedtuple
+from . import reader
+
+
+class Record:
+    """Abstract Fastq record object
+
+    Defines properties for accessing fastq record information:
+    :property name: name field
+    :property sequence: sequence field
+    :property name2: second name field
+    :property quality: quality field
+
+    Utilities for parsing read attributes:
+    :property average_quality: return the mean quality of FastqRecord
+    """
+    __slots__ = ['_data']
+
+    def __init__(self, record):
+        """
+        :param [str|bytes] record: list of four strings or bytes objects defining a
+          single fastq record
+        """
+        self._data = list(record)
```

Comment:
Is there a special reason why we are making a new list rather than using `record` as is, or are you just being careful? If there's a special reason, would be worth a comment.

Reply:
My docstring in line 21 is inaccurate; it should say "iterable" instead of list. This will often be read as a tuple. However, I require the ability to change the record, and so need it to be mutable. Hence the list conversion.